src/components/monitorAvailabilityModal/hooks/useMonitorAvailabilityModal.ts

```diff
@@ -29,6 +29,48 @@ const useMonitorAvailabilityModal = () => {
   const [sameAvailability, setSameAvailability] =
     useState<TWeekDayAvailabilityState>(SAME_AVAILABILITY_INITIAL_STATE);
 
+  const availabilities: TAvailability[] = useMemo(() => {
+    const selectedWeekDays = weekDayAvailability.filter(
+      (weekDay) => weekDay.isSelected
+    );
+
+    const availability: TAvailability[] = selectedWeekDays.map((weekDay) => ({
+      weekDay: weekDay.weekDay,
+      hours: [
+        {
+          start: sameAvailability.isSelected
+            ? HOUR_OPTIONS[sameAvailability.fromHourIndex]
+            : HOUR_OPTIONS[weekDay.fromHourIndex],
+          end: sameAvailability.isSelected
+            ? HOUR_OPTIONS[sameAvailability.toHourIndex]
+            : HOUR_OPTIONS[weekDay.toHourIndex],
+        },
+      ],
+    }));
+    return availability;
+  }, [weekDayAvailability, sameAvailability]);
+
+  const dayIsSelected: boolean = useMemo(() => {
+    return availabilities.length ? true : false;
+  }, [availabilities]);
```

Comment:
Pra ficar menos verboso:

```suggestion
  const dayIsSelected = useMemo(() => !!availabilities.length, [availabilities]);
```